### PR TITLE
iqtree: new port

### PR DESCRIPTION
**Pull request by ttung**
#### Description

Add iqtree 1.6.12

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.15
Xcode 11.4.1

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by ttung** [comment]
This [build](https://paste.macports.org/7998ac27294c) seems to have failed at an unrelated point.

File: science/iqtree/Portfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                        end, IQ-TREE can utilize multicore computers and distributed parallel \
+                        computing to speed up the analysis. IQ-TREE automatically performs \
+                        checkpointing to resume an interrupted analysis.
+
+homepage                http://www.iqtree.org/
+github.tarball_from     archive
+github.master_sites     ${github.homepage}/archive/
+distname                v${version}
+
+depends_build           port:eigen3 \
```
Comment (cjones051073) [inline science/iqtree/Portfile:35]: Please use 

depend_build-append

File: science/iqtree/Portfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                        rapid accumulation of phylogenomic data, leading to a need for \
+                        efficient phylogenomic software that can handle a large amount of \
+                        data and provide more complex models of sequence evolution. To this \
+                        end, IQ-TREE can utilize multicore computers and distributed parallel \
+                        computing to speed up the analysis. IQ-TREE automatically performs \
+                        checkpointing to resume an interrupted analysis.
+
+homepage                http://www.iqtree.org/
+github.tarball_from     archive
+github.master_sites     ${github.homepage}/archive/
```
Comment (cjones051073) [inline science/iqtree/Portfile:32]: This line shouldn’t be necessary, as if correctly setup the github PG should define it correctly

File: science/iqtree/Portfile [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+distname                v${version}
```
Comment (reneeotten) [inline science/iqtree/Portfile:33]: the `distname` line should be removed, which you can do when using the `github.setup` correctly (see above)

File: science/iqtree/Portfile [hunk 4]
```diff
@@ -0,0 +1,45 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem              1.0
+PortGroup               github 1.0
+PortGroup               cmake 1.1
+
+name                    iqtree
+github.setup            Cibiv IQ-TREE 1.6.12
```
Comment (reneeotten) [inline science/iqtree/Portfile:8]: this should become `github.setup            Cibiv IQ-TREE 1.6.12 v`

File: science/iqtree/Portfile [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                        checkpointing to resume an interrupted analysis.
+
+homepage                http://www.iqtree.org/
+github.tarball_from     archive
+github.master_sites     ${github.homepage}/archive/
+distname                v${version}
+
+depends_build-append    port:eigen3 \
+                        port:libomp
+
```
Comment (reneeotten) [inline science/iqtree/Portfile:37]: also from the build log you will see that it uses MacPorts zlib library (if present), so to get reproducible builds you'll likely want to add a dependency to it. 

Along the same lines, the build script checks (at least if I remember correctly last time I looked at this while installing it on Linux) for the presence of `MPI`. Again for reproducibility, this should be disabled to make sure it doesn't get picked up by chance; if you want you can likely provide a variant to enable it.

File: science/iqtree/Portfile (hunk 2 above)
Comment (ttung) [reply science/iqtree/Portfile:31]: Unfortunately, `github.tarball_from` set to `archive` sets `master_sites` to `                github.master_sites ${github.homepage}/archive/${git.branch}`.

File: science/iqtree/Portfile (hunk 1 above)
Comment (ttung) [reply science/iqtree/Portfile]: Fixed.

File: science/iqtree/Portfile (hunk 3 above)
Comment (ttung) [reply science/iqtree/Portfile]: Fixed.

File: science/iqtree/Portfile (hunk 4 above)
Comment (ttung) [reply science/iqtree/Portfile]: Fixed.

File: science/iqtree/Portfile (hunk 2 above)
Comment (reneeotten) [reply science/iqtree/Portfile:31]: this should work fine after you corrected the `github.setup` line, at least for me it does - perhaps you can try again

File: science/iqtree/Portfile (hunk 5 above)
Comment (reneeotten) [reply science/iqtree/Portfile:36]: there appears to be some weird stuff going on with the zlib dependency, for example in the build log you see message like:
`ld: warning: object file (/opt/local/lib/libz.a(gzclose.o)) was built for newer OSX version (10.14) than being linked (10.7)`

It appears that the the CMakeLists.txt file they do:
```
# to be compatible back to Mac OS X 10.7
add_definitions("-mmacosx-version-min=10.7")
set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.7")
```

I am by no means an expert on this, but I'd suspect that this doesn't align well with the MacPorts way of building things. Perhaps @cjones051073 has some better insights here?
```

File: science/iqtree/Portfile [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+homepage                http://www.iqtree.org/
+github.master_sites     ${github.homepage}/archive/
+
+depends_lib-append      port:eigen3 \
+                        port:libomp \
+                        port:zlib
+
+compiler.cpath-append   ${prefix}/include/libomp
+
```
Comment (reneeotten) [inline science/iqtree/Portfile:38]: the build system also adds `-std=c++11`, so you likely want to require a compiler that supports this; that can be done by adding  `compiler.cxx_standard   2011`

File: science/iqtree/Portfile (hunk 5 above)
Comment (cjones051073) [reply science/iqtree/Portfile:36]: yes, that could be problematic... 

If upstream don't provide an option to disable this 

a) Suggest to them to add an option, doing this unconditionally will interfere with build systems like MacPorts 
b) In the meantime, I would suggest using an `reinplace` to change it to the current SDK. something like

```
reinplace "s|mmacosx-version-min=10.7|mmacosx-version-min=${configure.sdk_version}|" ${worksrcpath}/CMakeLists.txt
```

Note that is untested, so make sure it works yourself...

File: science/iqtree/Portfile (hunk 2 above)
Comment (mf2k) [reply science/iqtree/Portfile:31]: I agree with everyone above. Manually setting `github.master_sites` should not be necessary.

File: science/iqtree/Portfile (hunk 5 above)
Comment (reneeotten) [reply science/iqtree/Portfile:36]: I just looked, an easier solution to me is to just patch (i.e., comment out/remove) everything except the last line from the CMakeLists.txt:
```
 elseif (APPLE)
        message("Target OS     : Mac OS X")
-       # to be compatible back to Mac OS X 10.7
-       if (IQTREE_FLAGS MATCHES "oldmac")
-               add_definitions("-mmacosx-version-min=10.5")
-               set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.5")
-       else()
-               add_definitions("-mmacosx-version-min=10.7")
-               set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.7")
-       endif()
+    ### to be compatible back to Mac OS X 10.7
+       ##if (IQTREE_FLAGS MATCHES "oldmac")
+       ##      add_definitions("-mmacosx-version-min=10.5")
+       ##      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.5")
+       ##else()
+       ##      add_definitions("-mmacosx-version-min=10.7")
+       ##      set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -mmacosx-version-min=10.7")
+       ##endif()
     SET(CMAKE_FIND_LIBRARY_SUFFIXES .a ${CMAKE_FIND_LIBRARY_SUFFIXES})
```

at least that works fine for me.

**Comment by reneeotten** [comment]
@ttung are you interested and willing to finish up this PR or do you want me to push the required changes?

**Comment by reneeotten** [comment]
okay, I  have pushed the remaining changes myself in an attempt to get this PR merged. Of note, since the final binary doesn't link to anything I have made the dependencies all `build` dependencies:

```
> otool -L /opt/local/bin/iqtree
/opt/local/bin/iqtree:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1252.250.1)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 400.9.4)
```

**Comment by reneeotten** [comment]
anyone else concerns/comments on this PR - if not, I will merge this tomorrow

**Pull request merged**